db/migrate/20230915105810_replace_correspondance_language_with_core_language.rb

```diff
@@ -0,0 +1,7 @@
+class ReplaceCorrespondanceLanguageWithCoreLanguage < ActiveRecord::Migration[6.1]
+  def change
+    Person.where(correspondence_language: nil).update_all(correspondence_language: :de)
```

Comment:
@carlobeltrame hier liegt das für den Moment vermeitlich grösste Problem: das Feld `correspondence_language` kann null sein, während `language` not null ist. Damit wird dieses Feld neu ein Pflichtfeld, was ja jeweils auf pfadipolitischer Ebene nicht unumstritten ist. Ich nehme hier auch gleich @simfeld dazu.

Was meint ihr?

Reply:
Ja gute Frage. Man könnte natürlich beginnen, den Kantonalverband der Haupt-Rolle jeder Person herauszusuchen, und anhand von dem die Sprache intelligent zu setzen... Kann man aber beliebig weit treiben weil es auch noch Kantone mit mehreren Sprachen gibt... Da hätte ich auch gerne einen Rat von @simfeld oder @Michael-Schaer wie weit wir hier gehen sollen.

Reply:
Was ist denn die Default-Korrespondenzsprache beim Versand von E-Mails wenn man keine gesetzt hat? Ich nehme an auch Deutsch? Insofern wäre für mich folgendes Vorgehen in Ordnung:

- Migration setzt per Default `language` auf DE
- PBS-Release-Notes kündigen dies an und man kann das somit verhindern indem man seine Korrespondenzsprache setzt vor dem Release

Reply:
Ich glaube die Mails achten bisher nicht auf die Korrespondenzsprache, könnte aber falsch liegen.
Deinen Vorschlag finde ich gut.

Reply:
Swing, Cosinus und ich haben uns heute auch noch abgesprochen. Wenn möglich, wäre ein Matching nach Kanton gut. Da die KVs Kantone zugeordnet haben, sollte ein Matching gut möglich sein. Hier der Vorschlag für die Aufteilung:

> **DE**
> Aargau Appenzell Innerrhoden Appenzell Ausserrhoden Bern Basel-Land Basel-Stadt 
> Glarus Graubünden Luzern Nidwalden Obwalden St. Gallen Schaffhausen Solothurn Schwyz Thurgau Uri Zug Zürich
> 
> **FR**
> Freiburg Genf Jura Waadt Wallis Neuenburg 
> 
> **IT**
> Tessin

Reply:
Ich habe das noch entsprechend in der Tech Spec ergänzt